app/questionnaire/value_source_resolver.py

```diff
@@ -116,6 +121,33 @@ def _resolve_answer_value_source(
 
         return answer_value
 
+    def _resolve_progress_value_source(
+        self, value_source: Mapping
+    ) -> ValueSourceEscapedTypes | ValueSourceTypes | None:
+        identifier = value_source["identifier"]
+        selector = value_source["selector"]
+        if selector == "section":
+            return self.progress_store.get_section_status(
+                section_id=identifier, list_item_id=None
+            )
+
+        if selector == "block":
+            if not self.location:
+                raise ValueError("location is required to resolve block progress")
+
+            section_id_for_block = self.schema.get_section_id_for_block_id(identifier)
+            if not self._is_block_on_path(identifier):
```

Comment:
The var should only be assigned after the path check, otherwise it is redundant as you would exit out anyway.

Reply:
Updated 👍